packages/tenant-process/src/services/tenantService.ts

```diff
@@ -1249,6 +1249,109 @@ export function tenantServiceBuilder(
 
       return tenantWithUpdatedKind;
     },
+
+    async m2mUpsertTenant(
+      m2mTenantSeed: tenantApi.M2MTenantSeed,
+      { authData, correlationId, logger }: WithLogger<AppContext>
+    ): Promise<Tenant> {
+      logger.info(
+        `Updating tenant with external id ${m2mTenantSeed.externalId} via m2m request`
+      );
+
+      const requesterTenant = await retrieveTenant(
+        authData.organizationId,
+        readModelService
+      );
+
+      const certifierId = retrieveCertifierId(requesterTenant.data);
+
+      const existingTenant = await readModelService.getTenantByExternalId(
+        m2mTenantSeed.externalId
+      );
+
+      assertTenantExists(
+        unsafeBrandId(
+          `${m2mTenantSeed.externalId.origin}/${m2mTenantSeed.externalId.value}`
+        ),
+        existingTenant
+      );
+
+      const attributesExternalIds = m2mTenantSeed.certifiedAttributes.map(
+        (externalId) =>
+          ExternalId.parse({
+            value: externalId.code,
+            origin: certifierId,
+          })
+      );
+
+      const attributesByExternalId =
+        await readModelService.getAttributesByExternalIds(
+          attributesExternalIds
+        );
+
+      for (const attributeToAssign of attributesExternalIds) {
+        if (
+          !attributesByExternalId.find(
+            (a) =>
+              a?.origin === attributeToAssign.origin &&
+              a?.code === attributeToAssign.value
+          )
+        ) {
```

Comment:
The array method `some` is more appropriate here

```suggestion
        if (
          !attributesByExternalId.some(
            (a) =>
              a?.origin === attributeToAssign.origin &&
              a?.code === attributeToAssign.value
          )
        ) {
```